include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -416,6 +416,49 @@ class argument_parser
         return *sub_parser;
     }
 
+    /*!\brief Checks whether the option distinguished by `id` was set on the command line by the user.
+     * \param[in] id The option identifier to search for (must not contain dashes!).
+     * \returns `true` if option identifier `id` was set on the command line by the user.
+     */
+    bool is_option_set(char const id)
+    {
+        if (!parse_was_called)
+            throw design_error("You can only asked which options have been set after calling the function `parse()`.");
+
+        auto constexpr allowed = is_alnum || is_char<'_'> || is_char<'@'>;
+
+        if (!allowed(id))
+            throw design_error("Short option identifiers may only contain alphanumeric characters, '_', or '@'.");
+
+        // we only need to search for an option before the `end_of_options_indentifier` (`--`)
+        auto end_of_options = std::find(cmd_arguments.begin(), cmd_arguments.end(), end_of_options_indentifier);
+        return detail::format_parse::find_option_id(cmd_arguments.begin(), end_of_options, id) != end_of_options;
+    }
+
+    // allows all types that are implicitly convertible to std::string
+    //!\overload
+    bool is_option_set(std::string const & id)
+    {
+        if (!parse_was_called)
+            throw design_error("You can only asked which options have been set after calling the function `parse()`.");
+
+        if (id.size() <= 1)
+            throw design_error("Long IDs must be longer than one character.");
+        else if (is_char<'-'>(id[0]))
+            throw design_error("First character of long ID cannot be '-'.");
+
+        auto constexpr allowed = is_alnum || is_char<'_'> || is_char<'@'> || is_char<'-'>;
+        std::for_each(id.begin(), id.end(), [&allowed] (char c)
+        {
+            if (!allowed(c))
+                throw design_error("Long identifiers may only contain alphanumeric characters, '_', '-', or '@'.");
+        });
+
+        // we only need to search for an option before the `end_of_options_indentifier` (`--`)
+        auto end_of_options = std::find(cmd_arguments.begin(), cmd_arguments.end(), end_of_options_indentifier);
+        return detail::format_parse::find_option_id(cmd_arguments.begin(), end_of_options, id) != end_of_options;
+    }
```

Comment:
Second, what happens if I call is_option_set("a")? is that really a design error? It would be a user error, right? But it should also just work. And is is_option_set('_') allowed as an option?
In addition, I see a lot of potential to DRY this function. For example, you could write a detail function that get's the option identifier always as a string and in addition the set of valid characters. Then you can only need to handle the really different things in the tpo-level functions.

Reply:
I thought about this again and I propose the following: Since we do enforce that `parse()` is called before this function can be used, I can now just check if the requested option identifier is in the list of `used_option_identifiers`. Those have already been tested in the `add_option` calls beforehand so I can omit all the testing completely?

Reply:
But you specify the option by a string, which could contain invalid characters, aren't you? So you would need to have some checks but you could resue what is already there to test them properly, which is probably the same code as done when adding the options?